gnome-image-installer/gnome-image-installer.c

```diff
@@ -140,28 +140,60 @@ read_sanitized_string (const gchar *filename,
   return sanitized;
 }
 
+/**
+ * read_install_ini:
+ * @path: path to directory holding image files
+ *
+ * Reads install.ini within @path, if it exists, and applies the locale
+ * specified within, if any. This file is created by the winstaller
+ * when creating a reformatter USB, containing the locale that the winstaller
+ * itself was displayed in.
+ *
+ * Historically, this file was also used to configure unattended installations,
+ * but this functionality is now split out into a separate file. If a locale is
+ * specified both install.ini and unattended.ini, the latter takes precedence.
+ */
 static void
-read_keys (const gchar *path)
+read_install_ini (const gchar *path)
 {
-  gchar *ini = g_build_path ("/", path, "install.ini", NULL);
-  GKeyFile *keys = g_key_file_new();
+  g_autofree gchar *ini = g_build_path ("/", path, "unattended.ini", NULL);
+  g_autoptr(GKeyFile) keys = g_key_file_new ();
+  g_autofree gchar *locale = NULL;
 
   if (g_key_file_load_from_file (keys, ini, G_KEY_FILE_NONE, NULL))
+    locale = g_key_file_get_string (keys, EOS_GROUP, LOCALE_KEY, NULL);
+
+  if (locale != NULL)
+    gis_language_page_preselect_language (locale);
+}
+
+static void
+read_unattended_ini (const gchar *path)
+{
+  g_autofree gchar *ini = g_build_path ("/", path, "unattended.ini", NULL);
+  g_autoptr(GisUnattendedConfig) config = NULL;
+  g_autoptr(GError) error = NULL;
+
+  config = gis_unattended_config_new (ini, &error);
+  if (config == NULL &&
+      !g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
```

Comment:
Nitpick: This condition would look a little clearer if it were `error != NULL && !g_error_matches (…)`.